sdk/inc/azure/core/az_platform.h

```diff
@@ -40,8 +45,13 @@ AZ_NODISCARD int64_t az_platform_clock_msec();
  *
  * @remarks The behavior is undefined when \p milliseconds is a non-positive value (0 or less than
  * 0).
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
+ * @retval #AZ_ERROR_DEPENDENCY_NOT_PROVIDED No platform implementation was supplied to support this
+ * function.
  */
-void az_platform_sleep_msec(int32_t milliseconds);
+AZ_NODISCARD az_result az_platform_sleep_msec(int32_t milliseconds);
```

Comment:
This is an unnecessary burden to the implementor of the PAL.

Reply:
@ahsonkhan are you referring to the AZ_NODISCARD or the az_result.  I agree that we shouldn't force AZ_NODISCARD onto the implementations.  I however do prefer the az_result return codes on the PAL.  The minimal burden is greatly outweighed by the fact we can consistently validate the API performed as expected

Reply:
Both.

> The minimal burden is greatly outweighed by the fact we can consistently validate the API performed as expected

I don't understand what you mean by this. What are we getting here?

Reply:
Ha, what I was thinking of, is that you Ahson was talking about the contrast of this variant to the proposal in https://github.com/Azure/azure-sdk-for-c/pull/1257, saying that out parameters and az_result are generally more cumbersome to implement. I didn't read it as if you are commenting on the AZ_NODISCARD.
Regarding AZ_NODISCARD, I do not think it is a burden on the implementor. It is an intentional burden on the user, i.e. us. So, if we ignore the return result, we get a warning treated as compilation error. For the implementor, I am not sure if they even have to put AZ_NODISCARD in the signature of their imlpementation - having it in this header alone is probably enough (I am not sure, because AZ_NODISCARD is really compiler-specific).

Reply:
OMHO, platform should not depend on az_core.  Only az_core should depend on some specific implementation.
Right now, in order to implement a platform, people need to get az_core, and people getting az_core need an az_platform.  I don't like how that look XD.  It works using cmake targets, but not sure if it would work without it (including pah directly)

Reply:
Interesting point on layering the dependency @vhvb1989